### PR TITLE
 runelite-client: Don't fail on invalid config values

### PR DESCRIPTION
**Pull request by Abextm**
As it is now, any plugin with an invalid config value fails to load. This makes it load with the default value instead.

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/Actor.java [hunk 1]
```diff
@@ -85,7 +85,7 @@
 	/**
 	 * Gets the server-side location of the actor.
 	 * <p>
-	 * This value is typically ahead of where the client renders and may
+	 * This value is typically ahead of where the client renders and is not
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/Actor.java]: "is not be affected" ?

File: runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java [hunk 2]
```diff
@@ -254,7 +254,15 @@ public String getConfiguration(String groupName, String key)
...
 		{
-			return (T) stringToObject(value, clazz);
+			try
+			{
+				return (T) stringToObject(value, clazz);
+			}
+			catch (Exception e)
+			{
+				log.warn("Unable to unmarshal {}.{} ", groupName, key, e);
+				return null;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java]: Why second return here? there is already one at bottom of the method

File: runelite-client/src/main/java/net/runelite/client/config/ConfigInvocationHandler.java [hunk 3]
```diff
@@ -78,16 +78,15 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
...
-
-			// objectValue automatically gets unboxed
-//			if (!objectValue.getClass().equals(returnType))
-//			{
-//				log.warn("Unable to convert return type for configuration item {}.{}: {}", group.keyName(), item.keyName(), returnType);
-//				return null;
-//			}
-
-			return objectValue;
+			try
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/config/ConfigInvocationHandler.java]: Add empty line above

File: runelite-client/src/main/java/net/runelite/client/config/ConfigInvocationHandler.java [hunk 4]
```diff
@@ -78,16 +78,16 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
...
-			return objectValue;
+			
+			try
+			{
+				return ConfigManager.stringToObject(value, returnType);
+			}
+			catch (Exception e)
+			{
+				log.warn("Unable to unmarshal {}.{} ", group.keyName(), item.keyName(), e);
+				return callDefaultMethod(proxy, method, args);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/config/ConfigInvocationHandler.java]: What if there is no default method?

**Pull request merged**